ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -350,10 +350,15 @@ class ( ShowLedgerState (LedgerTables l)
       , Eq (l EmptyMK)
       , Eq (LedgerTables l DiffMK)
       , Eq (LedgerTables l ValuesMK)
+      , Ord (TableKey l)
+      , Eq (TableValue l)
       ) => TableStuff (l :: LedgerStateKind) where
 
   data family LedgerTables l :: LedgerStateKind
 
+  type family TableKey l :: Type
+  type family TableValue l :: Type
```

Comment:
We can not define `TableKey l`, `TableValue l`, `projectMK` and `injectMK` for some ledger states. Though I don't know if this would work, one solution could be: 
* Define two type classes: one that mimicks the previous implementation of `TableStuff`, and one (let's call it `MoreTableStuff`) that adds things like `TableKey` on top. We could then write complete definitions of  `TableStuff` for ledger states like `Ledger.Dual`, and `MoreTableStuff` for other ledger states.

Reply:
For now let's only move the typeclass. We might never need to define those, but the suggestions can remain here and we can revisit it in the future, thanks for that